### PR TITLE
ocamlPackages.menhirLib: 20220210 -> 20230608

### PR DESCRIPTION
**Pull request by marsam**
###### Description of changes
Diff: https://gitlab.inria.fr/fpottier/menhir/-/compare/20220210...20230608

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build ocamlPackages.menhirLib

**Comment by vbgl** [comment]
@GrahamcOfBorg build ligo

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 238050` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ligo</li>
  </ul>
</details>
<details>
  <summary>51 packages built:</summary>
  <ul>
    <li>acgtk</li>
    <li>alt-ergo</li>
    <li>anders</li>
    <li>cmdstan</li>
    <li>coccinelle</li>
    <li>comby</li>
    <li>compcert</li>
    <li>compcert.doc</li>
    <li>compcert.lib</li>
    <li>compcert.man</li>
    <li>coqPackages.coq-elpi</li>
    <li>coqPackages.coq-lsp</li>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-algebra-tactics</li>
    <li>coqPackages.mathcomp-analysis</li>
    <li>coqPackages.mathcomp-classical</li>
    <li>coqPackages.trakt</li>
    <li>cvehound</li>
    <li>cvehound.dist</li>
    <li>dune-release</li>
    <li>easycrypt</li>
    <li>eff</li>
    <li>framac</li>
    <li>fstar</li>
    <li>heptagon</li>
    <li>jackline</li>
    <li>jasmin-compiler</li>
    <li>libbap</li>
    <li>liquidsoap</li>
    <li>obelisk</li>
    <li>ocamlformat (ocamlformat_0_24_1)</li>
    <li>ocamlformat_0_19_0</li>
    <li>ocamlformat_0_20_0</li>
    <li>ocamlformat_0_20_1</li>
    <li>ocamlformat_0_21_0</li>
    <li>ocamlformat_0_22_4</li>
    <li>ocamlformat_0_23_0</li>
    <li>python310Packages.bap</li>
    <li>python310Packages.bap.dist</li>
    <li>python311Packages.bap</li>
    <li>python311Packages.bap.dist</li>
    <li>reason</li>
    <li>sail-riscv-rv32</li>
    <li>sail-riscv-rv64</li>
    <li>satysfi</li>
    <li>soupault</li>
    <li>spark2014</li>
    <li>stanc</li>
    <li>stog</li>
    <li>wayland-proxy-virtwl</li>
    <li>why3</li>
  </ul>
</details>

**Comment by vbgl** [comment]
The recent update to `ligo` does not help. Ulrik, do you have an idea on how to solve this issue?

**Comment by ulrikstrid** [comment]
I assume the update today didn't help either, I'll try to see if I can find the issue and propose a patch. The company I work for is a sister company to the company behind LIGO so we have pretty good ties.

**Comment by ulrikstrid** [comment]
Ligo issue is here: https://gitlab.com/ligolang/ligo/-/issues/1947
It has some more information

**Comment by ulrikstrid** [comment]
https://gitlab.com/ligolang/ligo/-/merge_requests/2706/diffs

@marsam @vbgl

**Comment by ulrikstrid** [comment]
I took the liberty to push the LIGO patch, it didn't apply cleanly when I took it wholesale so I removed the `.opam` changes. We should be able to remove it on the next release.

File: pkgs/development/compilers/ligo/default.nix [hunk 1]
```diff
@@ -1,6 +1,7 @@
 { stdenv
 , lib
 , fetchFromGitLab
+, fetchpatch
```
Comment (vbgl) [inline pkgs/development/compilers/ligo/default.nix:4]: Is this needed?

Comment (ulrikstrid) [reply pkgs/development/compilers/ligo/default.nix:4]: Forgot to remove it, will fix

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 238050` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>305 packages built:</summary>
  <ul>
    <li>acgtk</li>
    <li>alt-ergo</li>
    <li>anders</li>
    <li>cmdstan</li>
    <li>coccinelle</li>
    <li>comby</li>
    <li>compcert</li>
    <li>compcert.doc</li>
    <li>compcert.lib</li>
    <li>compcert.man</li>
    <li>coqPackages.coq-elpi</li>
    <li>coqPackages.coq-lsp</li>
    <li>coqPackages.hierarchy-builder</li>
    <li>coqPackages.mathcomp-algebra-tactics</li>
    <li>coqPackages.mathcomp-analysis</li>
    <li>coqPackages.mathcomp-classical</li>
    <li>coqPackages.mathcomp-infotheo</li>
    <li>coqPackages.trakt</li>
    <li>cvehound</li>
    <li>cvehound.dist</li>
    <li>dune-release</li>
    <li>easycrypt</li>
    <li>eff</li>
    <li>framac</li>
    <li>fstar</li>
    <li>heptagon</li>
    <li>jackline</li>
    <li>jasmin-compiler</li>
    <li>libbap</li>
    <li>ligo</li>
    <li>liquidsoap</li>
    <li>obelisk</li>
    <li>ocamlPackages.alcotest-lwt</li>
    <li>ocamlPackages.alcotest-mirage</li>
    <li>ocamlPackages.arp</li>
    <li>ocamlPackages.async_js</li>
    <li>ocamlPackages.async_rpc_websocket</li>
    <li>ocamlPackages.async_smtp</li>
    <li>ocamlPackages.atd</li>
    <li>ocamlPackages.atdgen</li>
    <li>ocamlPackages.awa</li>
    <li>ocamlPackages.awa-mirage</li>
    <li>ocamlPackages.bap</li>
    <li>ocamlPackages.base64</li>
    <li>ocamlPackages.biocaml</li>
    <li>ocamlPackages.bistro</li>
    <li>ocamlPackages.bonsai</li>
    <li>ocamlPackages.bos</li>
    <li>ocamlPackages.brisk-reconciler</li>
    <li>ocamlPackages.brr</li>
    <li>ocamlPackages.ca-certs</li>
    <li>ocamlPackages.ca-certs-nss</li>
    <li>ocamlPackages.caqti</li>
    <li>ocamlPackages.caqti-async</li>
    <li>ocamlPackages.caqti-driver-mariadb</li>
    <li>ocamlPackages.caqti-driver-postgresql</li>
    <li>ocamlPackages.caqti-driver-sqlite3</li>
    <li>ocamlPackages.caqti-dynload</li>
    <li>ocamlPackages.caqti-lwt</li>
    <li>ocamlPackages.caqti-type-calendar</li>
    <li>ocamlPackages.carton</li>
    <li>ocamlPackages.carton-git</li>
    <li>ocamlPackages.carton-lwt</li>
    <li>ocamlPackages.checkseum</li>
    <li>ocamlPackages.cohttp</li>
    <li>ocamlPackages.cohttp-async</li>
    <li>ocamlPackages.cohttp-lwt</li>
    <li>ocamlPackages.cohttp-lwt-jsoo</li>
    <li>ocamlPackages.cohttp-lwt-unix</li>
    <li>ocamlPackages.cohttp-mirage</li>
    <li>ocamlPackages.cohttp-top</li>
    <li>ocamlPackages.cohttp_async_websocket</li>
    <li>ocamlPackages.cohttp_static_handler</li>
    <li>ocamlPackages.conduit-lwt-unix</li>
    <li>ocamlPackages.conduit-mirage</li>
    <li>ocamlPackages.containers</li>
    <li>ocamlPackages.containers-data</li>
    <li>ocamlPackages.cooltt</li>
    <li>ocamlPackages.ctypes_stubs_js</li>
    <li>ocamlPackages.curly</li>
    <li>ocamlPackages.dap</li>
    <li>ocamlPackages.decompress</li>
    <li>ocamlPackages.dedukti</li>
    <li>ocamlPackages.digestif</li>
    <li>ocamlPackages.dns</li>
    <li>ocamlPackages.dns-certify</li>
    <li>ocamlPackages.dns-cli</li>
    <li>ocamlPackages.dns-client</li>
    <li>ocamlPackages.dns-client-lwt</li>
    <li>ocamlPackages.dns-client-mirage</li>
    <li>ocamlPackages.dns-mirage</li>
    <li>ocamlPackages.dns-resolver</li>
    <li>ocamlPackages.dns-server</li>
    <li>ocamlPackages.dns-stub</li>
    <li>ocamlPackages.dns-tsig</li>
    <li>ocamlPackages.dnssec</li>
    <li>ocamlPackages.dolmen</li>
    <li>ocamlPackages.dose3</li>
    <li>ocamlPackages.eliom</li>
    <li>ocamlPackages.elpi</li>
    <li>ocamlPackages.email_message</li>
    <li>ocamlPackages.emile</li>
    <li>ocamlPackages.erm_xmpp</li>
    <li>ocamlPackages.ethernet</li>
    <li>ocamlPackages.flex</li>
    <li>ocamlPackages.frontc</li>
    <li>ocamlPackages.functoria</li>
    <li>ocamlPackages.gen_js_api</li>
    <li>ocamlPackages.git</li>
    <li>ocamlPackages.git-mirage</li>
    <li>ocamlPackages.git-paf</li>
    <li>ocamlPackages.git-unix</li>
    <li>ocamlPackages.github</li>
    <li>ocamlPackages.github-data</li>
    <li>ocamlPackages.github-jsoo</li>
    <li>ocamlPackages.github-unix</li>
    <li>ocamlPackages.graphql</li>
    <li>ocamlPackages.graphql-cohttp</li>
    <li>ocamlPackages.graphql-lwt</li>
    <li>ocamlPackages.graphql_parser</li>
    <li>ocamlPackages.graphql_ppx</li>
    <li>ocamlPackages.h2</li>
    <li>ocamlPackages.happy-eyeballs</li>
    <li>ocamlPackages.happy-eyeballs-lwt</li>
    <li>ocamlPackages.happy-eyeballs-mirage</li>
    <li>ocamlPackages.http-mirage-client</li>
    <li>ocamlPackages.imagelib</li>
    <li>ocamlPackages.incr_dom</li>
    <li>ocamlPackages.index</li>
    <li>ocamlPackages.irmin</li>
    <li>ocamlPackages.irmin-chunk</li>
    <li>ocamlPackages.irmin-containers</li>
    <li>ocamlPackages.irmin-fs</li>
    <li>ocamlPackages.irmin-git</li>
    <li>ocamlPackages.irmin-graphql</li>
    <li>ocamlPackages.irmin-http</li>
    <li>ocamlPackages.irmin-mirage</li>
    <li>ocamlPackages.irmin-mirage-git</li>
    <li>ocamlPackages.irmin-mirage-graphql</li>
    <li>ocamlPackages.irmin-pack</li>
    <li>ocamlPackages.irmin-test</li>
    <li>ocamlPackages.irmin-tezos</li>
    <li>ocamlPackages.irmin-watcher</li>
    <li>ocamlPackages.iter</li>
    <li>ocamlPackages.jingoo</li>
    <li>ocamlPackages.js_of_ocaml</li>
    <li>ocamlPackages.js_of_ocaml-compiler</li>
    <li>ocamlPackages.js_of_ocaml-lwt</li>
    <li>ocamlPackages.js_of_ocaml-ppx</li>
    <li>ocamlPackages.js_of_ocaml-ppx_deriving_json</li>
    <li>ocamlPackages.js_of_ocaml-toplevel</li>
    <li>ocamlPackages.js_of_ocaml-tyxml</li>
    <li>ocamlPackages.jwto</li>
    <li>ocamlPackages.lambda-term</li>
    <li>ocamlPackages.lambdapi</li>
    <li>ocamlPackages.letsencrypt</li>
    <li>ocamlPackages.letsencrypt-app</li>
    <li>ocamlPackages.letsencrypt-dns</li>
    <li>ocamlPackages.letsencrypt-mirage</li>
    <li>ocamlPackages.linol</li>
    <li>ocamlPackages.linol-lwt</li>
    <li>ocamlPackages.logs</li>
    <li>ocamlPackages.lustre-v6</li>
    <li>ocamlPackages.magic-trace</li>
    <li>ocamlPackages.mdx</li>
    <li>ocamlPackages.mdx.bin</li>
    <li>ocamlPackages.mdx.lib</li>
    <li>ocamlPackages.menhir</li>
    <li>ocamlPackages.menhirLib</li>
    <li>ocamlPackages.menhirSdk</li>
    <li>ocamlPackages.merlin</li>
    <li>ocamlPackages.metrics-lwt</li>
    <li>ocamlPackages.metrics-rusage</li>
    <li>ocamlPackages.metrics-unix</li>
    <li>ocamlPackages.mimic</li>
    <li>ocamlPackages.mimic-happy-eyeballs</li>
    <li>ocamlPackages.mirage</li>
    <li>ocamlPackages.mirage-block-combinators</li>
    <li>ocamlPackages.mirage-block-unix</li>
    <li>ocamlPackages.mirage-bootvar-xen</li>
    <li>ocamlPackages.mirage-channel</li>
    <li>ocamlPackages.mirage-crypto-ec</li>
    <li>ocamlPackages.mirage-crypto-pk</li>
    <li>ocamlPackages.mirage-crypto-rng</li>
    <li>ocamlPackages.mirage-crypto-rng-async</li>
    <li>ocamlPackages.mirage-crypto-rng-lwt</li>
    <li>ocamlPackages.mirage-crypto-rng-mirage</li>
    <li>ocamlPackages.mirage-flow-combinators</li>
    <li>ocamlPackages.mirage-flow-unix</li>
    <li>ocamlPackages.mirage-logs</li>
    <li>ocamlPackages.mirage-nat</li>
    <li>ocamlPackages.mirage-net-xen</li>
    <li>ocamlPackages.mirage-protocols</li>
    <li>ocamlPackages.mirage-runtime</li>
    <li>ocamlPackages.mirage-stack</li>
    <li>ocamlPackages.mirage-unix</li>
    <li>ocamlPackages.mirage-vnetif</li>
    <li>ocamlPackages.mirage-xen</li>
    <li>ocamlPackages.mirage.dev</li>
    <li>ocamlPackages.mldoc</li>
    <li>ocamlPackages.morbig</li>
    <li>ocamlPackages.mrmime</li>
    <li>ocamlPackages.mustache</li>
    <li>ocamlPackages.netchannel</li>
    <li>ocamlPackages.ocaml-print-intf</li>
    <li>ocamlPackages.ocaml-recovery-parser</li>
    <li>ocamlPackages.ocaml-vdom</li>
    <li>ocamlPackages.ocsigen-start</li>
    <li>ocamlPackages.ocsigen-toolkit</li>
    <li>ocamlPackages.ocsigen_server</li>
    <li>ocamlPackages.ocsipersist</li>
    <li>ocamlPackages.ocsipersist-pgsql</li>
    <li>ocamlPackages.ocsipersist-sqlite</li>
    <li>ocamlPackages.odate</li>
    <li>ocamlPackages.ojs</li>
    <li>ocamlPackages.opium</li>
    <li>ocamlPackages.oseq</li>
    <li>ocamlPackages.otoml</li>
    <li>ocamlPackages.otr</li>
    <li>ocamlPackages.paf</li>
    <li>ocamlPackages.paf-cohttp</li>
    <li>ocamlPackages.phylogenetics</li>
    <li>ocamlPackages.piaf</li>
    <li>ocamlPackages.piqi</li>
    <li>ocamlPackages.piqi-ocaml</li>
    <li>ocamlPackages.plotkicadsch</li>
    <li>ocamlPackages.ppx_css</li>
    <li>ocamlPackages.ppx_cstubs</li>
    <li>ocamlPackages.ppx_deriving_rpc</li>
    <li>ocamlPackages.ppx_deriving_yaml</li>
    <li>ocamlPackages.ppx_irmin</li>
    <li>ocamlPackages.ppx_python</li>
    <li>ocamlPackages.ppx_repr</li>
    <li>ocamlPackages.prettym</li>
    <li>ocamlPackages.printbox</li>
    <li>ocamlPackages.printbox-text</li>
    <li>ocamlPackages.progress</li>
    <li>ocamlPackages.promise_jsoo</li>
    <li>ocamlPackages.psmt2-frontend</li>
    <li>ocamlPackages.pyml</li>
    <li>ocamlPackages.pythonlib</li>
    <li>ocamlPackages.reason</li>
    <li>ocamlPackages.reason-native.console</li>
    <li>ocamlPackages.reason-native.dir</li>
    <li>ocamlPackages.reason-native.file-context-printer</li>
    <li>ocamlPackages.reason-native.fp</li>
    <li>ocamlPackages.reason-native.pastel</li>
    <li>ocamlPackages.reason-native.pastel-console</li>
    <li>ocamlPackages.reason-native.qcheck-rely</li>
    <li>ocamlPackages.reason-native.refmterr</li>
    <li>ocamlPackages.reason-native.rely</li>
    <li>ocamlPackages.reason-native.rely-junit-reporter</li>
    <li>ocamlPackages.rebez</li>
    <li>ocamlPackages.reperf</li>
    <li>ocamlPackages.repr</li>
    <li>ocamlPackages.resto-cohttp</li>
    <li>ocamlPackages.resto-cohttp-client</li>
    <li>ocamlPackages.resto-cohttp-self-serving-client</li>
    <li>ocamlPackages.resto-cohttp-server</li>
    <li>ocamlPackages.rpclib</li>
    <li>ocamlPackages.rpclib-lwt</li>
    <li>ocamlPackages.sail</li>
    <li>ocamlPackages.tar</li>
    <li>ocamlPackages.tar-unix</li>
    <li>ocamlPackages.tcpip</li>
    <li>ocamlPackages.telegraml</li>
    <li>ocamlPackages.tezos-base58</li>
    <li>ocamlPackages.thread-table</li>
    <li>ocamlPackages.tls</li>
    <li>ocamlPackages.tls-async</li>
    <li>ocamlPackages.tls-lwt</li>
    <li>ocamlPackages.tls-mirage</li>
    <li>ocamlPackages.toml</li>
    <li>ocamlPackages.tsort</li>
    <li>ocamlPackages.tyxml-lwd</li>
    <li>ocamlPackages.uring</li>
    <li>ocamlPackages.utop</li>
    <li>ocamlPackages.vg</li>
    <li>ocamlPackages.virtual_dom</li>
    <li>ocamlPackages.wayland</li>
    <li>ocamlPackages.webbrowser</li>
    <li>ocamlPackages.webmachine</li>
    <li>ocamlPackages.x509</li>
    <li>ocamlPackages.yaml</li>
    <li>ocamlPackages.yaml-sexp</li>
    <li>ocamlformat (ocamlformat_0_24_1)</li>
    <li>ocamlformat_0_19_0</li>
    <li>ocamlformat_0_20_0</li>
    <li>ocamlformat_0_20_1</li>
    <li>ocamlformat_0_21_0</li>
    <li>ocamlformat_0_22_4</li>
    <li>ocamlformat_0_23_0</li>
    <li>python310Packages.bap</li>
    <li>python310Packages.bap.dist</li>
    <li>python311Packages.bap</li>
    <li>python311Packages.bap.dist</li>
    <li>sail-riscv-rv32</li>
    <li>sail-riscv-rv64</li>
    <li>satysfi</li>
    <li>soupault</li>
    <li>spark2014</li>
    <li>stanc</li>
    <li>stog</li>
    <li>wayland-proxy-virtwl</li>
    <li>why3</li>
  </ul>
</details>

**Pull request merged**